src/Components/Notifications/ShowPushNotification.tsx

```diff
@@ -1,43 +1,41 @@
-import { useDispatch } from "react-redux";
-import { getNotificationData } from "../../Redux/actions";
-import { useEffect } from "react";
 import { DetailRoute } from "../../Routers/types";
+import useQuery from "../../Utils/request/useQuery";
+import routes from "../../Redux/api";
+import { NotificationData } from "./models";
 
 export default function ShowPushNotification({ id }: DetailRoute) {
-  const dispatch: any = useDispatch();
+  useQuery(routes.getNotificationData, {
+    pathParams: { id },
+    onResponse(res) {
+      if (res.data) {
+        window.location.href = resultUrl(res.data);
+      }
+    },
+  });
 
-  const resultUrl = async () => {
-    const res = await dispatch(getNotificationData({ id }));
-    const data = res.data.caused_objects;
-    switch (res.data.event) {
+  const resultUrl = (res: NotificationData) => {
+    const data = res?.caused_objects;
+    switch (res?.event) {
```

Comment:
@adriansliva 

1. This function can be moved outside the component right?
2. variable name `res` might be confused with `Response` instead of the data itself. How about this instead:

```suggestion
  const resultUrl = ({ caused_objects, event }: NotificationData) => {
    switch (event) {
```